plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -53,6 +38,78 @@ const closeDialog = ( context: ProductGalleryContext ) => {
 	}
 };
 
+const swiperContainer = document.querySelector(
+	'.wc-block-product-gallery-large-image'
+);
+
+const getSlideIndexByImageId = ( imageId: string ) => {
+	const slides = swiperContainer?.querySelectorAll( '.swiper-slide' );
+	let slideIndex = 0;
+
+	if ( slides ) {
+		slides.forEach( ( slide ) => {
+			const images = slide.querySelector(
+				'.wc-block-woocommerce-product-gallery-large-image__image'
+			);
+			const slideImageContext = images?.getAttribute( 'data-wc-context' );
+			const slideImageId = slideImageContext
+				? JSON.parse( slideImageContext ).imageId
+				: null;
+
+			if ( slideImageId === imageId ) {
+				const slideIndexAttr = slide.getAttribute(
+					'data-swiper-slide-index'
+				);
+
+				if ( slideIndexAttr !== null ) {
+					slideIndex = parseInt( slideIndexAttr, 10 );
+				}
+			}
+		} );
+	}
+
+	return slideIndex;
+};
+
+const getImageIdFromIndex = ( index: number ) => {
+	const imageIdContext = swiperContainer
+		?.querySelector(
+			`.swiper-slide[data-swiper-slide-index="${ index }"] img`
+		)
+		?.getAttribute( 'data-wc-context' );
+
+	const imageId = imageIdContext
+		? JSON.parse( imageIdContext ).imageId
+		: null;
+
+	return imageId;
+};
+
+const swiper = new Swiper( swiperContainer as HTMLElement, {
+	direction: 'horizontal',
+	loop: true,
+	on: {
+		beforeInit: () => {
+			if ( swiperContainer ) {
+				( swiperContainer as HTMLElement ).style.width =
+					( swiperContainer as HTMLElement ).clientWidth + 'px';
+			}
+		},
+		realIndexChange: ( swiperInstance ) => {
+			// Since we're not using the Swiper's pagination, we need to manually update the selected image for the pager.
+			const imageId = getImageIdFromIndex( swiperInstance.realIndex );
+
+			(
+				document.querySelector(
+					".wc-block-product-gallery-pager__pager-item[data-wc-context='" +
+						JSON.stringify( { imageId } ) +
+						"']"
+				) as HTMLButtonElement
+			 )?.click();
+		},
```

Comment:
I guess if went with the Swiper we'd eventually stop relying on iAPI? 🤔 I'm just thinking I wouldn't like us to keep two solutions in sync.